src/lit-element.ts

```diff
@@ -112,13 +129,13 @@ export class LitElement extends PropertiesMixin(HTMLElement) {
    * @param {*} changedProps Changing element properties
    * @param {*} prevProps Previous element properties
    */
-  protected _propertiesChanged(props: any, changedProps: any, prevProps: any) {
+  _propertiesChanged(props: any, changedProps: any, prevProps: any) {
     this.__isChanging = true;
     this.__isInvalid = false;
     super._propertiesChanged(props, changedProps, prevProps);
     const result = this.render(props);
```

Comment:
Where is render defined? I don't see it anywhere here.

Reply:
it's in here... it's abstract-ish